File: z-clients/javascript/src/models/authTokenUpdateIn.ts
// this file is @generated

export interface AuthTokenUpdateIn {
    namespace?: string | null;
    id: string;
    name?: string | null;
    expiryMs?: number | null;
    metadata?: { [key: string]: string } | null;
    scopes?: string[] | null;
    enabled?: boolean | null;
}

export const AuthTokenUpdateInSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): AuthTokenUpdateIn {
        return {
            namespace: object['namespace'],
            id: object['id'],
            name: object['name'],
            expiryMs: object['expiry_ms'],
            metadata: object['metadata'],
            scopes: object['scopes'],
            enabled: object['enabled'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: AuthTokenUpdateIn): any {
        return {
            'namespace': self.namespace,
            'id': self.id,
            'name': self.name,
            'expiry_ms': self.expiryMs,
            'metadata': self.metadata,
            'scopes': self.scopes,
            'enabled': self.enabled,
        };
    }
}

File: z-clients/rust/src/models/auth_token_update_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct AuthTokenUpdateIn {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub namespace: Option<String>,

    pub id: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub name: Option<String>,

    #[serde(
        rename = "expiry_ms",
        skip_serializing_if = "Option::is_none",
        with = "crate::duration_ms_serde::optional"
    )]
    pub expiry: Option<std::time::Duration>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub metadata: Option<std::collections::HashMap<String, String>>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub scopes: Option<Vec<String>>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub enabled: Option<bool>,
}

impl AuthTokenUpdateIn {
    pub fn new(id: String) -> Self {
        Self {
            namespace: None,
            id,
            name: None,
            expiry: None,
            metadata: None,
            scopes: None,
            enabled: None,
        }
    }

    pub fn with_namespace(mut self, value: impl Into<Option<String>>) -> Self {
        self.namespace = value.into();
        self
    }

    pub fn with_name(mut self, value: impl Into<Option<String>>) -> Self {
        self.name = value.into();
        self
    }

    pub fn with_expiry(mut self, value: impl Into<Option<std::time::Duration>>) -> Self {
        self.expiry = value.into();
        self
    }

    pub fn with_metadata(
        mut self,
        value: impl Into<Option<std::collections::HashMap<String, String>>>,
    ) -> Self {
        self.metadata = value.into();
        self
    }

    pub fn with_scopes(mut self, value: impl Into<Option<Vec<String>>>) -> Self {
        self.scopes = value.into();
        self
    }

    pub fn with_enabled(mut self, value: impl Into<Option<bool>>) -> Self {
        self.enabled = value.into();
        self
    }
}
